source/PlayerInfo.cpp

```diff
@@ -3320,7 +3320,7 @@ void PlayerInfo::RegisterDerivedConditions()
 	conditions["flagship model: "].ProvidePrefixed([this](const ConditionEntry &ce) -> bool {
 		if(!flagship)
 			return false;
-		return !ce.NameWithoutPrefix().compare(flagship->TrueModelName()); });
+		return ce.Name().ends_with(flagship->TrueModelName()); });
```

Comment:
This changes the behavior. `"flagship model: Marauder Arrow"` will return true if the player has an Arrow, since the condition name ends with the flagship name.